hadoop-mapreduce/src/test/java/org/apache/accumulo/hadoopImpl/mapreduce/InputFormatBuilderTest.java

```diff
@@ -0,0 +1,117 @@
+import org.junit.Before;
+import org.junit.Test;
+
+/*
+ This unit tests ClassLoaderContext and ExecuteHints functionality
+ */
+public class InputFormatBuilderTest {
+
+  private class InputFormatBuilderImplTest<T> extends InputFormatBuilderImpl<T> {
+    private String currentTable;
+    private SortedMap<String,InputTableConfig> tableConfigMap = new TreeMap<>();
+    private SortedMap<String,String> newHints = new TreeMap<>();
+
+    private InputFormatBuilderImplTest(Class callingClass) {
+      super(callingClass);
+    }
+
+    public InputFormatBuilder.InputFormatOptions<T> table(String tableName) {
+      this.currentTable = tableName;
+      tableConfigMap.put(currentTable, new InputTableConfig());
+      return this;
+    }
+
+    public InputFormatBuilder.InputFormatOptions<T> classLoaderContext(String context) {
+      tableConfigMap.get(currentTable).setContext(context);
+      return this;
+    }
+
+    private Optional<String> getClassLoaderContext() {
+      return tableConfigMap.get(currentTable).getContext();
+    }
+
+    public InputFormatBuilder.InputFormatOptions<T> executionHints(Map<String,String> hints) {
+      this.newHints.putAll(hints);
+      tableConfigMap.get(currentTable).setExecutionHints(hints);
+      return this;
+    }
+
+    private SortedMap<String,String> getExecutionHints() {
+      return newHints;
+    }
+  }
+
+  private InputTableConfig tableQueryConfig;
+  private InputFormatBuilderImplTest formatBuilderTest;
+
+  @Before
+  public void setUp() {
+    tableQueryConfig = new InputTableConfig();
+    formatBuilderTest = new InputFormatBuilderImplTest(InputFormatBuilderTest.class);
+    formatBuilderTest.table("test");
+  }
+
+  @Test
+  public void testInputFormatBuilder_ClassLoaderContext() {
+    String context = "classLoaderContext";
+
+    InputFormatBuilderImpl<JobConf> formatBuilder =
+        new InputFormatBuilderImpl<>(InputFormatBuilderTest.class);
+    formatBuilder.table("test");
+    formatBuilder.classLoaderContext(context);
+
+    Optional<String> classLoaderContextStr = tableQueryConfig.getContext();
+    assertTrue(classLoaderContextStr.toString().contains("empty")); // returns Optional.empty
+  }
+
+  @Test
+  public void testInputFormatBuilderImplTest_ClassLoaderContext() {
+    String context = "classLoaderContext";
+
+    formatBuilderTest.classLoaderContext(context);
+
+    Optional<String> classLoaderContextStr = formatBuilderTest.getClassLoaderContext();
+    assertEquals(classLoaderContextStr.get(), context);
+  }
+
+  @Test
+  public void testInputFormatBuilderImplTest_ExecuteHints() {
+    SortedMap<String,String> hints = new TreeMap<>();
+    hints.put("key1", "value1");
+    hints.put("key2", "value2");
+    hints.put("key3", "value3");
+
+    formatBuilderTest.executionHints(hints);
+
+    SortedMap<String,String> executionHints = formatBuilderTest.getExecutionHints();
+    assertEquals(executionHints.toString(), hints.toString());
```

Comment:
In an asserts statement, the expected values is the first argument. Please switch the order of the arguments.